### ISSUE TITLE
Client network socket disconnected

### ISSUE THREAD

**Issue by BogdanDarius** [mongodb-js/compass#2323]
#### Current Behavior

In version 1.26.1 SRV connection when ssl is not used works as expected.
In version 1.27.1 same configuration throws an error <b>Client network socket disconnected before secure TLS connection was established</b>.

#### Expected Behavior/Code

To connect to the cluster.

#### Environment

- node.js / npm versions: node v16.4.1, npm 7.17.0
- OS: Linux 5.12.14-arch1-1

#### Possible Solution

Maybe some ssl is not set properly?

#### Additional Context/Screenshots
![error](https://user-images.githubusercontent.com/25986436/124503389-204f4700-ddce-11eb-9258-48ff9d8c8bb6.png)

**Comment by mcasimir** [comment]
Hi @BogdanDarius, thanks for reporting this!

Could you paste here the connection string replacing username, password, hostnames and any information you don't want to share with placeholders?

It would be also great if you could answer few questions that would help us identifying the issue:

Are you trying to connect through a proxy / VPN / tunnel of sort?
Are you trying to connect to an Altlas cluster?
Are you able to reach all the nodes of the replica set from the host you use Compass?

Could you try to connect with `mongosh` (https://github.com/mongodb-js/mongosh/releases/tag/v0.15.5)?

Thanks a lot!

**Comment by BogdanDarius** [comment]
> Hi @BogdanDarius, thanks for reporting this!
> 
> Could you paste here the connection string replacing username, password, hostnames and any information you don't want to share with placeholders?
> 
> It would be also great if you could answer few questions that would help us identifying the issue:
> 
> Are you trying to connect through a proxy / VPN / tunnel of sort?
> Are you trying to connect to an Altlas cluster?
> Are you able to reach all the nodes of the replica set from the host you use Compass?
> 
> Could you try to connect with `mongosh` (https://github.com/mongodb-js/mongosh/releases/tag/v0.15.5)?
> 
> Thanks a lot!

As I stated in the issue report, it worked fine in 1.26.1 same connection string.

No I dont use proxy/VPN/tunnel.
I am trying to connect to a mongo db cluster that's inside a K8s cluster. I can connect to nodes one by one.
I get the same error with mongosh when using srv.

This doesnt work:
```bash 
mongodb+srv://MY_USERNAME:MY_PASSWORD@MY_DOMAIN/?authSource=admin&authMechanism=SCRAM-SHA-256&replicaSet=rs0&readPreference=primaryPreferred&appname=MongoDB%20Compass&ssl=false
```

This works (not using srv):
```bash
mongodb://MY_USERNAME:MY_PASSWORD@rs1.mongo.MY_DOMAIN:27017,rs0.mongo.MY_DOMAIN:27017,rs2.mongo.MY_DOMAIN:27017/?authSource=admin&authMechanism=SCRAM-SHA-256&replicaSet=rs0&readPreference=primaryPreferred&appname=MongoDB%20Compass&ssl=false
```

**Comment by mcasimir** [comment]
Hi @BogdanDarius we are tracking this on https://jira.mongodb.org/browse/COMPASS-4928, the work for a fix has been partially done and will be released soon.

**Comment by mcasimir** [comment]
Fixed in 1.28.1, feel free to reopen if upgrading does not solve the issue.
